core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -95,6 +95,37 @@ export class RubricsEditorComponent {
     return Boolean(this.editableExplanations[difficulty][index]);
   }
 
+  ismediumLevelExplanationLengthEqualsOne(difficulty: string): boolean {
+    if (difficulty === this.skillDifficultyMedium) {
+      return this.rubrics[1]._explanations.length === 1;
+    }
+    return false;
+  }
+
+  isMediumLevelExplanationValid(difficulty: string, data: string): boolean {
+    if (difficulty === this.skillDifficultyMedium) {
+      if (this.rubrics[1]._explanations.length > 1) {
+        return true;
+      }
+      return (data.length > 0);
+    }
+    return true;
+  }
+
+  isTotalExplanationsLengthReached(): boolean {
+    let totalEasyExplanations = this.rubrics[0]._explanations.length;
+    let totalMediumExplanations = this.rubrics[1]._explanations.length;
+    let totalHardExplanations = this.rubrics[2]._explanations.length;
```

Comment:
The frontend test is failing because one or more of the `this.rubrics[idx]` is set to `undefined`. If you add a print statement for each rubric level, what do you get?

Reply:
Thanks for putting up this PR, though!

Reply:
Thanks for the info, I added an if statement to resolve it.